### PR TITLE
Code Fixes and Improvements

### PR DESCRIPTION
**Pull request by JackSekoranja**
This PR achieves the following:
- Adds a 'publish' option to program plans, as requested by the original Issue (#19)
- Prevented submission if there are error messages in the Program form
- Unpublished Subplans can no longer be selected
- Subplans now restricted to the entered program year
   - Adding a 'subplan' vue component will automatically set up the 'year' field to update all components when the year changes
   - If the year changes when subplans are selected, the entered will need to be re-selected

### CODE REVIEW

File: cassdegrees/static/js/rules.js [hunk 1]
```diff
@@ -509,3 +533,10 @@ if (reqs.length > 0) {
         app.rules = parsed;
     }
 }
+
+
+function redrawVueComponents(){
```
Comment (j-selby) [inline cassdegrees/static/js/rules.js]: space after ()

File: cassdegrees/static/js/rules.js [hunk 2]
```diff
@@ -509,3 +533,10 @@ if (reqs.length > 0) {
...
+    for (var index in app.$children){
+        app.$children[index].do_redraw();
+    }
+}
```
Comment (j-selby) [inline cassdegrees/static/js/rules.js]: Newline

File: cassdegrees/templates/createprogram.html [hunk 3]
```diff
@@ -85,6 +85,9 @@
 
     <script>
         function submit_form(form_action) {
+            if(document.querySelector('.msg-error') !== null)
```
Comment (j-selby) [inline cassdegrees/templates/createprogram.html]: For consistency, enclose the return call in {}

File: cassdegrees/templates/widgets/rules.html [hunk 4]
```diff
@@ -13,6 +13,7 @@
     <fieldset v-if="!redraw">
         <div class="msg-error" v-if="non_unique_options">Options must be unique!</div>
         <div class="msg-error" v-if="inconsistent_units">Options have different unit values!</div>
+        <div class="msg-error" v-if="wrong_year_selected">Selected subplans must have same year as program</div>
```
Comment (j-selby) [inline cassdegrees/templates/widgets/rules.html]: Change to `Selected subplans must have same year as program!`

**Review by j-selby — CHANGES_REQUESTED** [review]
Looks good!

**Review by Daniel-Jang — APPROVED** [review]
I haven't extensively tested it but I had a quick look and play around and it seems good to me.

**Review by j-selby — APPROVED** [review]

**Pull request merged**